test/Writer/Factory/WriterFactoryTest.php

```diff
@@ -0,0 +1,290 @@
+        $this->assertEquals(['foo' => 'bar'], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__construct
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::createService
+     */
+    public function testCreateServiceWithValidRequestName()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ServiceLocatorInterface::class);
+
+        $pluginManager = $this->prophesize(AbstractPluginManager::class);
+        $pluginManager->getServiceLocator()->willReturn($container);
+        $pluginManager = $pluginManager->reveal();
+
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory->createService($pluginManager, 'invalid', InvokableObject::class);
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__construct
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::createService
+     * @expectedException \Zend\ServiceManager\Exception\InvalidServiceException
+     * @expectedExceptionMessage Zend\Log\Writer\Factory\WriterFactory requires that the requested
+     * name is provided on invocation; please update your tests or consuming container
+     */
+    public function testCreateServiceInvalidNames()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ServiceLocatorInterface::class);
+
+        $pluginManager = $this->prophesize(AbstractPluginManager::class);
+        $pluginManager->getServiceLocator()->willReturn($container);
+        $pluginManager = $pluginManager->reveal();
+
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory->createService($pluginManager, 'invalid', 'invalid');
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__invoke
+     */
+    public function testInvokeWithoutOptions()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ServiceLocatorInterface::class);
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, []);
+
+        // Assert
+        $this->assertInstanceOf(InvokableObject::class, $object);
+        $this->assertEquals([], $object->options);
+    }
+
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::__invoke
+     */
+    public function testInvokeWithInvalidFilterManagerAsString()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ContainerInterface::class);
+        $factory = new WriterFactory();
+
+        // Act
+        $object = $factory($container, InvokableObject::class, [
+            'filter_manager' => 'my_manager',
```

Comment:
If we make the changes as I suggested in the WriterFactory class above, this use case will now raise an exception (as an invalid manager is specified); this test will then need to change to assert against the expected exception.